### ISSUE TITLE
typescript decorators giving error in vs2015

### ISSUE THREAD

**Issue by pranaydutta89** [Microsoft/TypeScript#6476]
I am using typescript as my nodejs(ntvs) project and using decorators but its giving compile time error as ' Experimental support for decorators is a feature that is subject to change in a future release. Specify '--experimentalDecorators' to remove this warning.'

now since there is no tsconfig.json so i could not add the above paramtere which the compiler is asking ,and after searching for the issue on net , i changed the node.js project file and added the same parameter, then also still the issue persist, please help me with the issue.
![untitled](https://cloud.githubusercontent.com/assets/3788500/12316751/3569bc16-bab0-11e5-96a8-0bb332fc8773.png)

**Comment by mhegazy** [comment]
You will need to manually add `TypeScriptExperimentalDecorators` to your project file, by unload project, edit the project file, you should see something like:

```xml
  <PropertyGroup Condition="'$(Configuration)' == 'Debug'">
    <TypeScriptRemoveComments>false</TypeScriptRemoveComments>
    <TypeScriptSourceMap>true</TypeScriptSourceMap>
  </PropertyGroup>
```

add the new property tot he PropertyGroup:

```xml
<TypeScriptExperimentalDecorators>true</TypeScriptExperimentalDecorators>
```

for more infromation abotu MSBuild properties, please check: https://github.com/Microsoft/TypeScript/wiki/Setting-Compiler-Options-in-MSBuild-projects

**Comment by pranaydutta89** [comment]
i had already done that and this is not helping me out still the compiler is unhappy

**Comment by mhegazy** [comment]
Can you share the project file?

**Comment by pranaydutta89** [comment]
```
<?xml version="1.0" encoding="utf-8"?>
<Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003" ToolsVersion="4.0">
  <PropertyGroup>
    <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">11.0</VisualStudioVersion>
    <VSToolsPath Condition="'$(VSToolsPath)' == ''">$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)</VSToolsPath>
    <Name>edt_arch</Name>
    <RootNamespace>edt_arch</RootNamespace>
  </PropertyGroup>
  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <SchemaVersion>2.0</SchemaVersion>
    <ProjectGuid>3f62780c-8ace-47a8-82c7-53d1385e0ace</ProjectGuid>
    <ProjectHome>.</ProjectHome>
    <StartupFile>app.ts</StartupFile>
    <StartWebBrowser>False</StartWebBrowser>
    <SearchPath>
    </SearchPath>
    <WorkingDirectory>.</WorkingDirectory>
    <OutputPath>.</OutputPath>
    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
    <ProjectTypeGuids>{3AF33F2E-1136-4D97-BBB7-1795711AC8B8};{9092AA53-FB77-4645-B42D-1CCCA6BD08BD}</ProjectTypeGuids>
    <ProjectView>ProjectFiles</ProjectView>
    <TypeScriptSourceMap>true</TypeScriptSourceMap>
    <TypeScriptModuleKind>CommonJS</TypeScriptModuleKind>
    <EnableTypeScript>true</EnableTypeScript>
    <StartWebBrowser>false</StartWebBrowser>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
    <DebugSymbols>true</DebugSymbols>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
    <DebugSymbols>true</DebugSymbols>
  </PropertyGroup>
  <ItemGroup>
    <Content Include="sqlScripts.sql" />
    <Content Include="views\index.html" />
    <Content Include="views\templates\master.html" />
    <Content Include="views\templates\private\master.html" />
    <Content Include="views\templates\private\privateHome.html" />
    <Content Include="views\templates\private\refreshMicroService.html" />
    <Content Include="views\templates\private\registerMicroService.html" />
    <TypeScriptCompile Include="app.ts" />
    <Content Include="package.json" />
    <Content Include="README.md" />
    <TypeScriptCompile Include="config.ts" />
    <TypeScriptCompile Include="constants.ts" />
    <TypeScriptCompile Include="controllers\microServicesRoutesController.ts" />
    <TypeScriptCompile Include="controllers\portalControllers.ts" />
    <TypeScriptCompile Include="controllers\registerControllers.ts" />
    <TypeScriptCompile Include="Scripts\typings\express\express.d.ts" />
    <TypeScriptCompile Include="Scripts\typings\interfaces\domain.d.ts" />
    <TypeScriptCompile Include="Scripts\typings\node\node.d.ts" />
  </ItemGroup>
  <ItemGroup>
    <Folder Include="controllers\" />
    <Folder Include="Scripts\typings\interfaces\" />
    <Folder Include="views\" />
    <Folder Include="Scripts\" />
    <Folder Include="Scripts\typings\" />
    <Folder Include="Scripts\typings\express\" />
    <Folder Include="Scripts\typings\node\" />
    <Folder Include="views\contents\scripts\" />
    <Folder Include="views\contents\" />
    <Folder Include="views\templates\" />
    <Folder Include="views\src\" />
    <Folder Include="views\templates\private\" />
    <Folder Include="views\templates\public\" />
  </ItemGroup>
  <ItemGroup>
    <Compile Include="views\contents\scripts\angular-ui-router.min.js" />
    <Compile Include="views\contents\scripts\angular.min.js" />
    <Compile Include="views\src\controllers.js" />
    <Compile Include="views\src\init.js" />
    <Compile Include="views\src\services.js" />
  </ItemGroup>
  <!-- Do not delete the following Import Project.  While this appears to do nothing it is a marker for setting TypeScript properties before our import that depends on them. -->
  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'">
    <TypeScriptToolsVersion>1.7</TypeScriptToolsVersion>
    <TypeScriptTarget>ES5</TypeScriptTarget>
    <TypeScriptJSXEmit>None</TypeScriptJSXEmit>
    <TypeScriptCompileOnSaveEnabled>True</TypeScriptCompileOnSaveEnabled>
    <TypeScriptNoImplicitAny>False</TypeScriptNoImplicitAny>
    <TypeScriptModuleKind>CommonJS</TypeScriptModuleKind>
    <TypeScriptRemoveComments>False</TypeScriptRemoveComments>
    <TypeScriptOutFile />
    <TypeScriptOutDir />
    <TypeScriptGeneratesDeclarations>False</TypeScriptGeneratesDeclarations>
    <TypeScriptNoEmitOnError>False</TypeScriptNoEmitOnError>
    <TypeScriptMapRoot />
    <TypeScriptSourceRoot />
    <TypeScriptEmitDecoratorMetadata>True</TypeScriptEmitDecoratorMetadata>
   <TypeScriptExperimentalDecorators>true</TypeScriptExperimentalDecorators>
  </PropertyGroup>
  <Import Project="$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\TypeScript\Microsoft.TypeScript.targets" Condition="False" />
  <Import Project="$(VSToolsPath)\Node.js Tools\Microsoft.NodejsTools.targets" />
  <ProjectExtensions>
    <VisualStudio>
      <FlavorProperties GUID="{349c5851-65df-11da-9384-00065b846f21}">
        <WebProjectProperties>
          <UseIIS>False</UseIIS>
          <AutoAssignPort>True</AutoAssignPort>
          <DevelopmentServerPort>0</DevelopmentServerPort>
          <DevelopmentServerVPath>/</DevelopmentServerVPath>
          <IISUrl>http://localhost:48022/</IISUrl>
          <NTLMAuthentication>False</NTLMAuthentication>
          <UseCustomServer>True</UseCustomServer>
          <CustomServerUrl>http://localhost:1337</CustomServerUrl>
          <SaveServerSettingsInUserFile>False</SaveServerSettingsInUserFile>
        </WebProjectProperties>
      </FlavorProperties>
      <FlavorProperties GUID="{349c5851-65df-11da-9384-00065b846f21}" User="">
        <WebProjectProperties>
          <StartPageUrl>
          </StartPageUrl>
          <StartAction>CurrentPage</StartAction>
          <AspNetDebugging>True</AspNetDebugging>
          <SilverlightDebugging>False</SilverlightDebugging>
          <NativeDebugging>False</NativeDebugging>
          <SQLDebugging>False</SQLDebugging>
          <ExternalProgram>
          </ExternalProgram>
          <StartExternalURL>
          </StartExternalURL>
          <StartCmdLineArguments>
          </StartCmdLineArguments>
          <StartWorkingDirectory>
          </StartWorkingDirectory>
          <EnableENC>False</EnableENC>
          <AlwaysStartWebServerOnDebug>False</AlwaysStartWebServerOnDebug>
        </WebProjectProperties>
      </FlavorProperties>
    </VisualStudio>
  </ProjectExtensions>
</Project>
```

**Comment by pranaydutta89** [comment]
@mhegazy  please provide me with a solution .

**Comment by mhegazy** [comment]
sorry for that. looks like a bug.

**Comment by mhegazy** [comment]
@pranaydutta89, to work around the issue for now, please edit your project file and replace:
```XML
<PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Debug|AnyCPU'">
```
with
```XML
<PropertyGroup Condition="'$(Configuration)' == 'Debug'">
```

**Comment by pranaydutta89** [comment]
@mhegazy  u snippet did the trick thanks for help

**Comment by paulvanbrenk** [comment]
Checked in a fix, so we write the TypeScript properties in a more logical location.

**Comment by Dzivo** [comment]
This is still a problem none of the solution works i still get errors that i didnt set experimental decorators  to true

**Comment by mhegazy** [comment]
@Dzivo this is a fairly old issue. do you mind filing a new issue and provide information about what version of TS/VS are you using, also a sample project would be helpful.
